src/proto/TheOpenNetwork.proto

```diff
@@ -55,13 +55,72 @@ message Transfer {
     bool bounceable = 8;
 }
 
+message WalletTransferData {
+    // Wallet version
+    WalletVersion wallet_version = 1;
+
+    // Recipient address
+    string dest = 2;
+
+    // Amount to send in nanotons
+    uint64 amount = 3;
+
+    // Message counter (optional, 0 by default used for the first deploy)
+    // This field is required, because we need to protect the smart contract against "replay attacks"
+    // Learn more: https://ton.org/docs/develop/smart-contracts/guidelines/external-messages
+    uint32 sequence_number = 4;
+
+    // Send mode (optional, 0 by default)
+    // Learn more: https://ton.org/docs/develop/func/stdlib#send_raw_message
+    uint32 mode = 5;
+
+    // Expiration UNIX timestamp (optional, now() + 60 by default)
+    uint32 expire_at = 6;
+
+    // If the address is bounceable
+    bool bounceable = 8;
+}
+
+// same as transfer, but uses WalletTransferData
+message SimpleTransfer {
+    WalletTransferData wallet_transfer_data = 1;
+
+    // Transfer comment message (optional, empty by default)
+    string comment = 2;
```

Comment:
As I can see, the only difference between `WalletTransferData` and `Transfer` is the `comment` property. But `comment` is in the both `SimpleTransfer` and `JettonTransfer` messages.
I agree that having the `Transfer transfer_data = 1;` field in `JettonTransfer` may confuse the end-user, so `WalletTransferData` makes more sense in that case.
But it has a few drawbacks:
1. We need to deprecate `Transfer` message and ask existing end-user apps to use `SimpleTransfer` instead
2. If we plan to remove `Transfer` in the next iterations, we'll still need to support both approaches.
3. If we plan to leave both `Transfer` and `SimpleTransfer`, it will be complicated to figure out what approach the end-user needs to follow

I would remove `WalletTransferData` and `SimpleTransfer`, and refactor `JettonTransfer` this way:
```proto
message JettonTransfer {
  // dest in WalletTransferData means contract address of sender's jetton wallet
  Transfer wallet_transfer_data = 1;
  // ...
}
```

@Milerius what do you think?

Reply:
The reason for creation of simple transfer is that basically not every transfer can contain comment. For example, there are no comments for NFT transfer. Also comments are handled differently in simple transfer and jetton transfer. 
I see that the solution with `WalletTransferData` also have it's drawbacks. 
Please let me know which one would you like to choose finally

Reply:
No preference for me, we can rework that when we move it to rust too

Reply:
@Milerius do I understand correctly that we'll ask end-user apps to use `SimpleTransfer` instead of `Transfer`, after that remove it at the next iterations?

Reply:
Sorry, re-read everything I also agree with ur comment, make more sense for me to avoid code duplication

Reply:
I will refactor as @satoshiotomakan suggests then

Reply:
@satoshiotomakan refactored, please check

Reply:
Sorry for pushes, twitsted a bit so that you can see changes quickly: https://github.com/trustwallet/wallet-core/compare/4bf8803524061ce60980122a0b1a9a77fc808e5a..e721f9a01b73c5c373487dfacd32765a07f50d27